cocotb/drivers/__init__.py

```diff
@@ -278,7 +281,8 @@ def __init__(self, entity, name, clock, valid_generator=None):
             valid_generator (generator): a generator that yields tuples  of
                                         (valid, invalid) cycles to insert
         """
-        BusDriver.__init__(self, entity, name, clock)
+        BusDriver.__init__(self, entity, name, clock, **kwargs)
+        valid_generator = kwargs.get("valid_generator", None)
```

Comment:
Typically you'd write this using `pop`, to prevent `BusDriver.__init__` receiving a `valid_generator` argument that was not intended for it:
```python
valid_generator = kwargs.pop("valid_generator", None)
BusDriver.__init__(self, entity, name, clock, **kwargs)
```

This is how matplotlib handles this style of forwarded. An advantage of this approach is it behaves the same as the python 3 syntax,
```python
def __init__(self, entity, name, clock, *, valid_generator=None, **kwargs):
```

Reply:
You're right, using `pop` for the config is also used in other cocotb files - will change this for consistency